### ISSUE TITLE
Cannot import Postman variables from Swagger spec

### ISSUE THREAD

**Issue by sutdisi** [a85/POSTMan-Chrome-Extension#2951]
Repros in Postman both chrome app and native app. Native app version 4.10.7

We have different environments with different hostnames and using either http or https. When I try to parameterize these while importing Postman I have two issues. 1) Variable gets corrupted 2) Postman always sticks http in front of any URL making it impossible to parameterize.

Repro Steps:
1) Import this modified minimal petstore swagger definition:
```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "name": "Swagger API Team"
    },
    "license": {
      "name": "MIT"
    }
  },
  "host": "{{environment_url}}{{portNumber}}",
  "basePath": "/api",
  "schemes": [
    "{{scheme}}"
  ],
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "paths": {
    "/pets": {
      "get": {
        "description": "Returns all pets from the system that the user has access to",
        "produces": [
          "application/json"
        ],
        "responses": {
          "200": {
            "description": "A list of pets.",
            "schema": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/Pet"
              }
            }
          }
        }
      }
    }
  },
  "definitions": {
    "Pet": {
      "type": "object",
      "required": [
        "id",
        "name"
      ],
      "properties": {
        "id": {
          "type": "integer",
          "format": "int64"
        },
        "name": {
          "type": "string"
        },
        "tag": {
          "type": "string"
        }
      }
    }
  }
}
```
2) Check the GET API call in Postman

RESULT:
http://{{environment_urlpostman_variable_close_d/B{{portNumber}}/api/pets

EXPECTED:
{{scheme}}://{{environment_url}}{{portNumber}}/api/pets

If there was a way to remove default http from the final URL I could use {{environment_url}} with http/https in it like so:
{{environment_url}}{{portNumber}}/api/pets

**Comment by mattcrwi** [comment]
I resolved this by exporting the postman library and hand editing the configuration and re-importing. This is a pain that will happen every time I change my swagger and want to re-import.

**Comment by mslancashire** [comment]
This issue still exists I ran into it today when trying to import all our API Services into POSTMAN during our CLI build process.

We have set host to be "api.{{EnviromentPart}.{{DomainPart}}" which then transforms into "api.{{EnviromentPartpostman_variable_close_d/B.{{DomainPart}}".

After looking through your code in Git Hub found where the issue is occurring just not sure of the fix.

https://github.com/postmanlabs/swagger2-postman2/blob/61da825b7c8e9808656133989f3788ecf493deb0/lib/helpers.js#L178

This is where it replaces {{  and }} into replacement variables (POSTMAN_VARIABLE_OPEN_DB and POSTMAN_VARIABLE_CLOSE_DB), then pushes the result through the JS decodeURI and Nodes url.resolve.  Its the result of these two methods that result in the replacement back to {{ and }} not working.

Nodes url.resolve url encodes chars, see the spec.
https://nodejs.org/api/url.html#url_url_resolve_from_to.

I almost got a work around for this working by instead of using {{ and }} using the encoded versions instead %7B%7B and %7D respectively.  However for some reason I end up with an extra / at the start. So "api.%7B%7BEnvironmentPart%7D%7D.%7B%7BDomainPart%7D%7D" becomes "api/{{EnvironmentPart}}.{{DomainPart}}". So close.....

Any idea of when this can be looked at?

**Comment by abhijitkane** [comment]
@sutdisi @mattcrwi @mslancashire Apologies for the delay on this. This has been fixed in Postman >7.2
